unit/goto-programs/goto_program_instruction.cpp

```diff
@@ -0,0 +1,116 @@
+/*******************************************************************\
+ Module: Unit tests for goto_program::validate
+
+ Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+#include <goto-programs/goto_function.h>
+#include <goto-programs/validate_goto_model.h>
+#include <testing-utils/use_catch.h>
+#include <util/arith_tools.h>
+
+#include <util/c_types.h>
+#include <util/std_code.h>
+
+SCENARIO(
+  "Validation of well-formed instruction",
+  "[core][goto-programs][validate]")
+{
+  GIVEN("A function with one instruction")
+  {
+    symbol_tablet symbol_table;
+    symbolt fun_symbol;
+    irep_idt fun_name = "foo";
+    fun_symbol.name = fun_name;
+
+    goto_functiont goto_function;
+    auto &instructions = goto_function.body.instructions;
+
+    goto_programt::instructiont instruction;
+    instruction.make_end_function();
+    instructions.push_back(instruction);
+    instructions.back().source_location.id("id_any_valid_id");
```

Comment:
The source_locationt ireps don't have an ID.

Reply:
Done.